### DISCUSSION TITLE
Unrealistic rotamers for D-amino acids with aromatic side chains. Side chains appear to be skewed/smeared in output pdb.

### DISCUSSION THREAD

**Discussion by tsamdin** [RosettaCommons/rosetta#245] [Rosetta] [answered]
I'm getting unrealistic rotamers for D-aromatic side chains. I made sure to include the LimitAromaChi2 task operation - but I'm not sure if this applies to rotamers for D-amino acids? The rings appear skewed/smeared - with unrealistic bond lengths?

Rotamers for L-aromatic side chains in the rest of the protein (not shown) appear to be realistic, so I believe the issue is limited to just the D-amino acids.

Any recommendations on how I can address this behavior? 

![dstrand](https://github.com/user-attachments/assets/406eaf9f-b273-474b-9a59-8596e062ae45)

**Comment by everyday847** [comment]
Some observations:

- The D-amino acids are constructed by an exact mirror image flip of the L-amino acids. All the ideal bond lengths and angles are identical, up to PDB rounding errors.
- The rotamer libraries are similarly flipped, and only have to do with dihedral angle values.

So, two questions:
- Are you packing in the context of `relax`? Are you certain this is *not* happening during the minimization steps?
- Are you running a non-ideal relax variant (i.e., either letting bond lengths and angles vary in internal coordinate relax, or an explicitly Cartesian relax)?

If so, my highest suspicion is that this is an issue in `cart_bonded` where some parameters (particularly for rings) are not cloned to D-amino acids.

**Reply by tsamdin** [reply]
I'm going through my XML script and looking more closely at my relax/design/pack movers. One thing that stands out to me immediately is how I set up my MinPackMover. I had nonideal set to 1 - which from the documentation allows "nonideal (minimization) : open up the bond-angle- and bond-length DOFs to minimization". I'm going to set nonideal="0" and see if that makes a difference. 

<MinPackMover name="MPM" scorefxn="r15_cart" **nonideal="1"** off_rotamer_pack="1"
     task_operations="use_input_rotamer,extrarot,no_repack_L_protein,no_design_L_protein,L_only_design,D_only_design,D_buried,L_buried,D_strand_buried_resnum,D_strand_exposed_resnum,no_cys,no_Gly_CYS_DCYS,conservative,no_ASP_DASP,aromachi" 
        />

**Reply by tsamdin** [reply]
Setting nonideal=0 did not prevent the generation of unrealistic D-amino acids with rings.

I found a link to an old forum post that describes a similar issue [https://forum.rosettacommons.org/node/3620](url). I was using ref2015_cart_cst.wts for the score fxn used by MinPackMover, so I'll try switching to the recommended score12minpack.wts. - hopefully this is the fix. 

Also in my setup of the MinPackMover, I don't have cartesian turned on for MinPackMover, so I'll give that a go as well.

**Reply by everyday847** [reply]
Yeah, I'd systematically take movers out and add them back in to narrow down which component we need to dig deeper on. Anything that minimizes is suspect. (Similarly, any protocol that sets up a scorefunction or minmover using some unusual logic -- if it ignores some input flag it should consider or whatever.)

**Comment by vmullig** [comment]
To check whether this is a packing issue or a minimizing issue, try
replacing the MinPackMover with the PackRotamersMover.  The output should
just be on-rotamer conformations then.

**Comment by tsamdin** [comment]
So I tracked down the issue to a Fastrelax mover that followed minpackmover, where I had bondlengths and bondangles=1.  I wasn't using dualspace or cartesian in this mover. 

What I did to resolve my problem was turn on dualspace, and change the score function to one that was setup for cartesian minimization. As a follow up question, should I also add 2 rounds of cartesian minimization as described in the relax documentation?

 _Use the Dualspace protocol for dihedral and cartesian minimization as described by Conway et al. Do 3 FastRelax cycles of internal coordinate relax followed by two cycles of Cartesian relax - cart_bonded energy term is required, pro_close energy term should be turned off, and use of -relax::minimize_bond_angles is recommended.  The -nonideal flag can be used for this._

[https://docs.rosettacommons.org/docs/latest/application_documentation/structure_prediction/relax](url)

**Comment by everyday847** [comment]
You don't have to, but you can. That's the status of innumerable protocols.
That paper showed better results on a benchmark that may or may not be
related to your scientific use case.

**Reply by tsamdin** [reply]
Thank you!

**Comment by roccomoretti** [comment, marked as answer]
> What I did to resolve my problem was turn on dualspace, and change the score function to one that was setup for cartesian minimization. 

It was the changing of the scorefunction which did it. You can do flexible bond lengths and angles with the regular (atom tree) minimizer/relax, but since the regular scorefunctions don't contain any terms which restrain those DOFs, you can get messes from it. The `cart_bonded` term adds restraints for bond lengths and angles. That's very important for Cartesian (and dualspace) minimization, but it's also needed for flexible bond length and angle minimization in internal coordinate space as well. (So 'cart' here is a bit of a historical misnomer.)

**Reply by tsamdin** [reply]
Thank you!
